scripts/get-suppliers-with-copyable-services.py

```diff
@@ -36,30 +40,48 @@
     new_framework_slug = arguments['<new-framework>']
     copy_from_framework_slug = arguments['<copy-from-framework>']
 
-    suppliers_on_old_framework = data_api_client.find_framework_suppliers_iter(copy_from_framework_slug)
+    suppliers_on_old_framework = data_api_client.find_framework_suppliers_iter(
+        copy_from_framework_slug,
+        agreement_returned=True
+    )
+
+    print(f"Building CSV with supplier accounts")
+
+    if not os.path.exists(OUTPUT_DIR):
+        print("Creating {} directory".format(OUTPUT_DIR))
+        os.makedirs(OUTPUT_DIR)
+
+    with open(os.path.join(OUTPUT_DIR, f"supplier-accounts-{new_framework_slug}.csv"), "w", newline='') as file:
+        writer = csv.writer(file, delimiter=',')
+        writer.writerow(["supplier_id", "email_address", "password"])
+
+        count = 0
+
+        for supplier in suppliers_on_old_framework:
+            if count >= SUPPLIER_LIMIT:
+                break
 
-    print("supplier_id,email_address,password")
+            supplier_id = supplier['supplierId']
+            for service in data_api_client.find_draft_services_iter(supplier_id, framework=new_framework_slug):
+                data_api_client.delete_draft_service(service['id'])
 
-    for count, supplier in enumerate(suppliers_on_old_framework):
-        if count >= SUPPLIER_LIMIT:
-            break
+            try:
+                data_api_client.remove_supplier_declaration(92197, new_framework_slug)
+                data_api_client.set_supplier_framework_application_company_details_confirmed(
+                    supplier_id, new_framework_slug, False
+                )
+            except HTTPError as e:
+                if not (e.status_code == 404 and "has not registered interest" in e.message):
+                    raise
 
-        supplier_id = supplier['supplierId']
-        for service in data_api_client.find_draft_services_iter(supplier_id, framework=new_framework_slug):
-            data_api_client.delete_draft_service(service['id'])
+            try:
+                user = next(u for u in data_api_client.find_users_iter(supplier_id=supplier_id) if u['active'])
+            except StopIteration:
+                continue
 
-        try:
-            data_api_client.remove_supplier_declaration(92197, new_framework_slug)
-            data_api_client.set_supplier_framework_application_company_details_confirmed(
-                supplier_id, new_framework_slug, False
-            )
-        except HTTPError as e:
-            if not (e.status_code == 404 and "has not registered interest" in e.message):
-                raise
+            count += 1
 
-        try:
-            user = next(u for u in data_api_client.find_users_iter(supplier_id=supplier_id) if u['active'])
-        except StopIteration:
-            continue
+            print(f"Suppliers found: {count}/{SUPPLIER_LIMIT}", end='\r')
```

Comment:
Why the `end` here?

(not a problem that needs to be fixed - I'm just curious)

Reply:
It is a counter so I can see the progress. The end means it will just overwrite the current count with the updated count and means we don't get 1000 lines in the terminal.

It looks something like this:
<img width="302" alt="Screenshot 2021-11-18 at 11 12 23" src="https://user-images.githubusercontent.com/58297459/142404993-4e356c55-1a05-4d3c-b4df-964bec60b6f1.png">

Reply:
That's neat - I didn't know about that. Thanks!